File: ui/react/src/action/EraseRows.ts
import TableAction from "./TableAction";
import Action from "./Action";

class EraseRows extends TableAction {
    public rowIndex: number;
    public nRows: number;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const tableName: string = input.tableName;
        const rowIndex: number = input.rowIndex;
        const nRows: number = input.nRows;
        return new EraseRows(sheetName, tableName, rowIndex, nRows);
    }

    constructor(
        sheetName: string,
        tableName: string,
        rowIndex: number,
        nRows: number) {

        super(sheetName, tableName);
        this.rowIndex = rowIndex;
        this.nRows = nRows;
    }

    public wrapper(): Action {
        const action = new Action();
        action.eraseRows = this;
        return action;
    }

}

export default EraseRows;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/EraseRows.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.api.action.Action;

public class EraseRows extends TableAction implements Action {
    private int rowIndex;
    private int nRows;

    public EraseRows() {
    }

    public EraseRows(String sheetName, String tableName, int rowIndex, int nRows) {
        super(sheetName, tableName);
        this.rowIndex = rowIndex;
        this.nRows = nRows;
    }

    public int getRowIndex() {
        return rowIndex;
    }

    public void setRowIndex(int rowIndex) {
        this.rowIndex = rowIndex;
    }

    public int getnRows() {
        return nRows;
    }

    public void setnRows(int nRows) {
        this.nRows = nRows;
    }
}
